### ISSUE TITLE
Default setter is overwritten by A.CallTo method

### ISSUE THREAD

**Issue by MrVermond** [FakeItEasy/FakeItEasy#1329]
Hi all i just found an issue in the current  version 4.5.

```c#

public interface ISomeObject
{
        object Width{ get; set; }
}

public interface ITestIt{

        ISomeObject Property{get; set; }
}


public void TestForSetter ()
{
            var object1 = A.Fake<ISomeObject>();
            var object2 = A.Fake<ISomeObject>();
            var object3 = A.Fake<ISomeObject>();

            var test = A.Fake<ITestIt>();
            test.Property = object1;
            var bla1 = test.Property;
            A.CallTo(() => test.Property).Returns(object2);
            var bla = test.Property;
            test.Property = object3; 
            Assert.That(test.Property, Is.EqualTo(object3)); //this does not work anymore
 }
```

**Comment by blairconrad** [comment]
Hi, @MrVermond. I'm trying to work through this, but am unable so far. What are `object1`, `object2`, and `object3`? I tried making them new `SomeType`s (which also are missing a definition, so I used an empty class for that type), but I still don't know what `ActitveConfiguration` is.

Are you able to produce a small example that compiles and reproduces the problem, or failing that, expand on the question and provide more detailed information about what you expected to happen and what actually happened, ideally with the actual FakeItEasy failure?

**Comment by MrVermond** [comment]
just updated the code sorry was my mistake

**Comment by blairconrad** [comment]
I'm trying, but the code is still not so good:

1. no `;` after the setters - easy for me to work around
2. no `SomeType` definition. Even if I make one up, assigning a fake `ISomeObject` to `ITest.Property` fails because there's no conversion between an `ISomeObject` and a `SomeType`.
3. Still no definition for `ActiveConfiguration`.

**Comment by MrVermond** [comment]
hmm check again

**Comment by thomaslevesque** [comment]
I think this behavior is as designed. See [the documentation](http://fakeiteasy.readthedocs.io/en/stable/default-fake-behavior/#readwrite-properties):

> By default, any overrideable property that has both a set and get accessor, **and that has not been explicitly configured to behave otherwise**, behaves like you might expect. Setting a value and then getting the value returns the value that was set.

(emphasis mine)

So when you explicitly configure the getter with `A.CallTo`, it overrides the default property behavior.

**Comment by MrVermond** [comment]
Hmm in previous versions it worked. I just was expecting the same behaviour in the latest version.
If so im fine (would be a bigger refactoring of our code).

**Comment by blairconrad** [comment]
@MrVermond, as @thomaslevesque says, the behaviour is as designed, and has been like this for a very long time. I'm seeing the same behaviour in 4.4.0. Can you tell us what version you're upgrading from?

**Comment by MrVermond** [comment]
2.2.0 :)

**Comment by blairconrad** [comment]
https://github.com/FakeItEasy/FakeItEasy/issues/962

**Comment by MrVermond** [comment]
ah ok, thank you for the info. We need defenetely to update :)

**Comment by blairconrad** [comment]
A pleasure, @MrVermond. Sorry for your surprise.

Oh, and above I said 

> was fixed in 3.0.0

That implies it was a bug. We'd determined at the time that the old behaviour wasn't a bug, just something that we didn't think should continue. I should've said "changed in 3.0.0".
